docs/en/extn-developer-FAQ.md

```diff
@@ -0,0 +1,93 @@
+# Extension Developer FAQ
+
+**# # # Work in Progress # # #** 
+
+## How does the non-OS SDK structure execution
+
+Apologies for this long description, but none of this is documented by
+Espressif. So this summarises our understanding of how **non-OS SDK** execution
+model works based on the Espressif-supplied examples and SDK documentation, plus
+various posts on the Espressif BBS and other forums, and an examination of the
+BootROM code.
+
+The ESP8266 boot ROM contains a set of primitive tasking and dispatch functions
+which are also used by the SDK. In this model, execution units are either:
+
+-   **INTERRUPT SERVICE ROUTINES (ISRs)** which are declared and controlled
+    through the `ets_isr_attach()` and other `ets_isr_*` and `ets_intr_*`
+    functions. ISRs can be defined on a range of priorities, where a higher
```

Comment:
I think it is 2+NMI = 3 but my point here is that it is more than 1.  Interrupts can be interrupted.